From 40b3e85ef570201de84e998264059d63e0d36fee Mon Sep 17 00:00:00 2001
From: Yong Zhao <yong.zhao@amd.com>
Date: Fri, 12 Jan 2018 14:40:06 -0500
Subject: [PATCH 4630/5855] drm/amdkfd: Disable Raven KFD support on pre-4.16
 kernel

Because some important IOMMU driver fixes were only available after
kernel 4.16, we should do so.

Change-Id: Ibf4e33ce8da3a9731dbcbb1ced9c8d0639424d34
Signed-off-by: Yong Zhao <yong.zhao@amd.com>
---
 drivers/gpu/drm/amd/amdgpu/amdgpu_amdkfd.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/gpu/drm/amd/amdgpu/amdgpu_amdkfd.c b/drivers/gpu/drm/amd/amdgpu/amdgpu_amdkfd.c
index 62ffe0e..49045b4 100644
--- a/drivers/gpu/drm/amd/amdgpu/amdgpu_amdkfd.c
+++ b/drivers/gpu/drm/amd/amdgpu/amdgpu_amdkfd.c
@@ -98,6 +98,12 @@ void amdgpu_amdkfd_device_probe(struct amdgpu_device *adev)
 		break;
 	case CHIP_VEGA10:
 	case CHIP_RAVEN:
+#if LINUX_VERSION_CODE < KERNEL_VERSION(4, 16, 0)
+		if (adev->asic_type == CHIP_RAVEN) {
+			dev_dbg(adev->dev, "DKMS installed kfd does not support Raven for kernel < 4.16\n");
+			return;
+		}
+#endif
 		kfd2kgd = amdgpu_amdkfd_gfx_9_0_get_functions();
 		break;
 	default:
-- 
2.7.4

